### PR TITLE
Modify compactor script to accommodate more memory options

### PR DESCRIPTION
**Pull request by SravanthiAshokKumar**
The compactor script is modified to accommodate more memory options that will be passed to it as a config file

## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: scripts/compactor_runner.py [hunk 1]
```diff
@@ -62,13 +62,18 @@ class CommandBuilder(object):
...
 
-    def derive_xmx_value(self, diskBacked):
+    def derive_xmx_value(self, diskBacked, compactor_config):
         try:
             mem_bytes = os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES')
             mem_mb = int(mem_bytes / (1024. ** 2))
             xmx = max(512, int(mem_mb * 0.04))
             if diskBacked is True:
-                xmx = max(self._config.xmx_min_in_mb, int(mem_mb * float(self._config.xmx_perc)/100))
+                if xmx > 1000 and "Large" in compactor_config["MemoryOptions"]:
```
Comment (PavelZaytsev) [inline scripts/compactor_runner.py:71]: Maybe use global vars?

**Review by PavelZaytsev — APPROVED** [review]

**Review by chetangudisagar — APPROVED** [review]
`compactor_config["MemoryOptions"]`
Where is this value coming from? Are you planning to add the field `MemoryOptions` in `corfu-compactor-config.yml` file?

File: scripts/compactor_runner.py [hunk 2]
```diff
@@ -62,13 +62,18 @@ class CommandBuilder(object):
...
+                    xmx = compactor_config["MemoryOptions"]["Large"]
+                elif xmx > 512 and xmx < 1000 and "Medium" in compactor_config["MemoryOptions"]:
+                    xmx = compactor_config["MemoryOptions"]["Medium"]
+                else:
+                    xmx = max(self._config.xmx_min_in_mb, int(mem_mb * float(self._config.xmx_perc)/100))
```
Comment (Lujie1996) [inline scripts/compactor_runner.py:76]: Change it to `compactor_config["MemoryOptions"]["Small"]`?

Comment (SravanthiAshokKumar) [reply scripts/compactor_runner.py:80]: Added a line for small and retained the else part in case there's no Xmx value passed in the config file

**Comment by SravanthiAshokKumar** [comment]
> 

It'll be passed in the config script - path of the config script should be passed while invoking this script

**Review by hisundar — APPROVED** [review]

**Pull request merged**